dateparser/search/search.py

```diff
@@ -86,10 +86,15 @@ def split_if_not_parsed(self, item, original):
                 possible_splits.extend(self.split_by(item, original, splitter))
         return possible_splits
 
-    def parse_item(self, parser, item, translated_item, parsed, need_relative_base):
+    def parse_item(self, parser, item, translated_item, parsed, need_relative_base, language=None):
         relative_base = None
-        item = item.replace('ngày', '')
-        item = item.replace('am', '')
+
+        # Replacing language specific words with special meaning
+        if language == "de":
+            item = item.replace('am', '')
```

Comment:
```suggestion
        if language == "de":
            # Replace "am" because "am 8" means "on the 8th" but "8 am" actually means "8 am"
            item = item.replace('am', '')
```

Reply:
Hmmmm... Maybe we can use regex (`re.sub`) to remove it only when preceding a number? :thinking:

Reply:
Hi @noviluni, I have made some can you please check and recommend